src/sites/twitch-twilight/modules/directory/community.js

```diff
@@ -0,0 +1,314 @@
+			directory {
+				... on Community {
+					streams {
+						edges {
+							node {
+								createdAt
+								type
+								broadcaster {
+									profileImageURL(width: 70)
+								}
+							}
+						}
+					}
+				}
+			}
+		}`);
+
+		this.ChannelCard = this.fine.define(
+			'community-channel-card',
+			n => n.props && n.props.streamNode
+		);
+
+		this.apollo.registerModifier('GamePage_Game', res => this.router.current.name === 'dir-community' && this.modifyStreams(res), false);
+
+		this.on('settings:changed:show-channel-avatar', value => {
+			this.css_tweaks.toggleHide('profile-hover-game', value === 2);
+			this.router.current.name === 'dir-community' && this.ChannelCard.forceUpdate();
+		});
+
+		this.on('settings:changed:directory.following.uptime', () => this.router.current.name === 'dir-community' && this.ChannelCard.forceUpdate());
+	}
+
+	modifyStreams(res) { // eslint-disable-line class-methods-use-this
+		const newStreams = [];
+
+		const edges = res.data.directory.streams.edges;
+		for (let i = 0; i < edges.length; i++) {
+			const edge = edges[i];
+			const node = edge.node;
+			
+			const s = node.viewersCount = new Number(node.viewersCount || 0);
+			s.profileImageURL = node.broadcaster.profileImageURL;
+			s.createdAt = node.createdAt;
+
+			newStreams.push(edge);
+		}
+		res.data.directory.streams.edges = newStreams;
+		return res;
+	}
+
+	onEnable() {
+		this.GameHeader.ready((cls, instances) => {
+			if (this.router.current.name === 'dir-community') {
+				for(const inst of instances) this.updateButtons(inst);
+			}
+		});
+
+		this.ChannelCard.ready((cls, instances) => {
+			if (this.router.current.name === 'dir-community') {
+				this.apollo.ensureQuery(
+					'GamePage_Game',
+					'data.directory.streams.edges.0.node.createdAt'
+				);
+
+				for(const inst of instances) this.updateChannelCard(inst);
+			}
+		});
+
+		this.ChannelCard.on('update', inst => this.router.current.name === 'dir-community' && this.updateChannelCard(inst), this);
+
+		this.ChannelCard.on('mount', inst => this.router.current.name === 'dir-community' && this.updateChannelCard(inst), this);
+
+		this.ChannelCard.on('unmount', inst => this.router.current.name === 'dir-community' && this.updateUptime(inst), this);
+
+		this.css_tweaks.toggleHide('profile-hover-game', this.settings.get('directory.following.show-channel-avatar') === 2);
+	}
+
+	updateUptime(inst) {
+		const container = this.fine.getHostNode(inst);
+		const card = container && container.querySelector && container.querySelector('.tw-card');
+
+		if (this.settings.get('directory.following.uptime') === 0) {
+			if (inst.update_timer !== undefined) {
+				clearInterval(inst.update_timer);
+				inst.update_timer = undefined;
+			}
+
+			if (inst.uptime_element !== undefined) {
+				inst.uptime_element.remove();
+				inst.uptime_element_span = inst.uptime_element = undefined;
+			}
+		} else {
+			if (inst.update_timer === undefined) {
+				inst.update_timer = setInterval(
```

Comment:
I think this is leaking timers - you only call clearInterval() when the feature is disabled, but you should also be doing that when the element is removed (getHostNode() is null?)